consul_exporter.go

```diff
@@ -360,6 +375,7 @@ func main() {
 	kingpin.Flag("consul.key-file", "File path to a PEM-encoded private key used with the certificate to verify the exporter's authenticity.").Default("").StringVar(&opts.keyFile)
 	kingpin.Flag("consul.server-name", "When provided, this overrides the hostname for the TLS certificate. It can be used to ensure that the certificate name matches the hostname we declare.").Default("").StringVar(&opts.serverName)
 	kingpin.Flag("consul.timeout", "Timeout on HTTP requests to consul.").Default("200ms").DurationVar(&opts.timeout)
+	kingpin.Flag("consul.request-limit", "Limit the maximum number of concurrent requests to consul.").Default("0").IntVar(&opts.requestLimit)
```

Comment:
Do you have a suggestion for a default which prevents overloads out-of-the-box? I'd not leave it it unbounded by default.

Reply:
I think it's better to set the default value to 1. This way the default load is minimal for the consul. 
    If the user wants to collect metrics faster, they can increase this value according to their environment.
    In my test environment , about one hundred services, and the request-limit is 1, the exporter can still complete the metrics collection within 200ms.

Reply:
@grobie